### DISCUSSION TITLE
Change basic_credential crate to use crypto provider instead of reimplementing methods to generate keypair and sign

### DISCUSSION THREAD

**Discussion by erskingardner** [openmls/openmls#1621] [Ideas]
Right now the basic_credential crate only supports RustCrypto and reimplements methods to generate a new keypair and to sign. This should be changed so that it can reuse the methods implemented by the Crypto Provider. 

For example, I'm working on bringing MLS messaging to Nostr and I've created a openmls_nostr_crypto provider that brings support for schnorr signatures over secp256k1 via the `secp256k1` crate. The methods in the basic_credential are mirroring functionality from the crypto provider traits but doesn't use those methods.

**Comment by franziskuskiefer** [comment]
The `basic_credential` crate was written mostly to have a credential to test with. But I can see that it would be useful to have some easy to use default credential types in the repo here. I'd be happy to take contributions to make it better and use the crypto providers.
